### ISSUE TITLE
Always render hidden input, even when no values are selected

### ISSUE THREAD

**Issue by everdimension** [JedWatson/react-select#1441]
_TL;DR:_
This will help to validate forms using native html form validation APIs.

---

Preamble:

`React-select` sends only the value in the `onChange` callback. That makes it much harder to work with `react-select` than with simple `input` elements. That's because if I have several `react-select` components in my form I cannot create a generic `handleChange` handler for these comonents because I don't have a way to know which component was changed (no access to something like `event.target.name`).

But ok. I can create special handlers for each component quite easily.

**The problem** is when it comes to validation:

`React-select` conveniently renders `input[type=hidden]` elements. I can access them via `form.elements[name]` getter. But... not so conveniently afterall.
Turns out, `react-select` only renders these hidden inputs only when some value is selected (https://github.com/JedWatson/react-select/blob/v1.0.0-rc.1/src/Select.js#L945)

That's a problem. When nothing is selected, I don't have an elegant way of validating my form via [`html5 form validation`](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Forms/Data_form_validation) api because the necessary inputs are absent.

My suggestion is to always render the hidden input fields. It seems to me that it would also make it easier to handle the `required` prop.

**Comment by silvenon** [comment]
Possibly a duplicate of #1400.

**Comment by bladey** [comment]
Hello -

In an effort to sustain the `react-select` project going forward, we're closing old issues / pull requests.

We understand this might be inconvenient but in the best interest of supporting the broader community we have to direct our limited efforts to maintain the latest version.

If you feel this issue / pull request is still relevant and you'd like us to review it, please leave a comment and we'll do our best to get back to you.
